backend/src/kitconcept/intranet/serializers/blocks.py

```diff
@@ -1,33 +1,16 @@
 from kitconcept.intranet.interfaces import IBrowserLayer
-from plone.restapi.behaviors import IBlocks
-from plone.restapi.blocks import iter_block_transform_handlers
-from plone.restapi.blocks import visit_blocks
-from plone.restapi.interfaces import IBlockFieldSerializationTransformer
-from plone.restapi.interfaces import IFieldSerializer
-from plone.restapi.serializer.converters import json_compatible
-from plone.restapi.serializer.dxfields import DefaultFieldSerializer
+from plone.restapi.types.adapters import JSONFieldSchemaProvider
+from plone.restapi.types.interfaces import IJsonSchemaProvider
 from plone.schema import IJSONField
 from zope.component import adapter
+from zope.interface import Interface
 from zope.interface import implementer
 
-import copy
 
-
-@adapter(IJSONField, IBlocks, IBrowserLayer)
-@implementer(IFieldSerializer)
-class BlocksJSONFieldSerializer(DefaultFieldSerializer):
-    def __call__(self):
-        value = copy.deepcopy(self.get_value())
-
-        blocksValue = value.get("blocks", value)
-
-        if self.field.getName() == "blocks" or "blocks" in value:
-            for block in visit_blocks(self.context, blocksValue):
-                new_block = block.copy()
-                for handler in iter_block_transform_handlers(
-                    self.context, block, IBlockFieldSerializationTransformer
-                ):
-                    new_block = handler(new_block)
-                block.clear()
-                block.update(new_block)
-        return json_compatible(value)
+@adapter(IJSONField, Interface, IBrowserLayer)
+@implementer(IJsonSchemaProvider)
+class CustomJSONFieldSchemaProvider(JSONFieldSchemaProvider):
+    def get_schema(self):
+        schema = super().get_schema()
+        schema.update(self.field.json_schema)
+        return schema
```

Comment:
You don't need this since you're explicitly setting the widget to object_list and using schemaName. The frontend is not using the schema from the backend.

Reply:
I know, I left it un-wired, I wondered if it was worth to leave it here in case we need it in the future. Maybe add it to p.restapi by default won't hurt either...

I will remove it, to keep things tidy.